PM-25311: CXF Map all custom fields to a CustomFieldsCredential

abergs opened the pull request:
## 🎟️ Tracking
https://bitwarden.atlassian.net/browse/PM-25311
<!-- Paste the link to the Jira or GitHub issue or otherwise describe / point to where this change is coming from. -->

## 📔 Objective

This PR is a quick stab at mapping all custom fields on a cipher to a single CustomFieldCredential.

I'm not 100% sure about how to parse this from the spec:

> If the [exporting provider](https://fidoalliance.org/specs/cx/cxf-v1.0-ps-20250814.html#exporting-provider) allows custom fields to be added to items but does not have a grouping concept, it SHOULD use this object without setting the label or id fields.

From: https://fidoalliance.org/specs/cx/cxf-v1.0-ps-20250814.html#dict-custom-fields



<!-- Describe what the purpose of this PR is, for example what bug you're fixing or new feature you're adding. -->

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation
  team

## 🦮 Reviewer guidelines

<!-- Suggested interactions but feel free to use (or not) as you desire! -->

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

abergs committed:
Map all custom fields to a customfieldsCredential

Hinton commented:
> If the [exporting provider](https://fidoalliance.org/specs/cx/cxf-v1.0-ps-20250814.html#exporting-provider) allows custom fields to be added to items but does not have a grouping concept, it SHOULD use this object without setting the label or id fields.

Means we should leave ID and label of the group empty.

Hinton commented on crates/bitwarden-exporters/src/cxf/export.rs:
```diff
@@ -56,6 +57,35 @@ impl TryFrom<Cipher> for Item {
+                        (Some(name), Some(value)) => Some(EditableField {
+                            id: None,
+                            label: Some(name),
+                            value: EditableFieldString(value),
```
**issue:** We support text, hidden, checkbox and linked.
- Text -> `String`
- Hidden -> `ConcealedString`
- Checkbox -> `Boolean`
- Linked -> not sure, keep string maybe but it's probably an extension.

**suggestion:** Add some logic to `editable_field` to map `Field` -> `EditableField` properly.

abergs replied:
@Hinton I've taken a stab at it. I'm can't say I feel confident in the way it came together though.

abergs committed:
Map the types more properly

abergs committed:
fmt

abergs commented:
Update: @Hinton patched the underlying CXF library to add support for the missing types. I've updated our dependency to his PR and used the types in the mapping.

Also added support for the importing case where we previously just used string.

Hinton commented on crates/bitwarden-exporters/src/cxf/import.rs:
```diff
@@ -49,7 +49,41 @@ fn custom_fields_to_fields(custom_fields: &CustomFieldsCredential) -> Vec<Field>
+            }
+            credential_exchange_format::EditableFieldValue::WifiNetworkSecurityType(field) => {
+                create_field(field, None::<String>)
+            }
```
**suggestion:** If we import `EditableFieldValue` in this file we can get this on single lines.

```rust
fn custom_fields_to_fields(custom_fields: &CustomFieldsCredential) -> Vec<Field> {
    custom_fields
        .fields
        .iter()
        .map(|field_value| match field_value {
            EditableFieldValue::String(f) => create_field(f, None::<String>),
            EditableFieldValue::ConcealedString(f) => create_field(f, None::<String>),
            EditableFieldValue::Boolean(f) => create_field(f, None::<String>),
            EditableFieldValue::Date(f) => create_field(f, None::<String>),
            EditableFieldValue::YearMonth(f) => create_field(f, None::<String>),
            EditableFieldValue::SubdivisionCode(f) => create_field(f, None::<String>),
            EditableFieldValue::CountryCode(f) => create_field(f, None::<String>),
            EditableFieldValue::WifiNetworkSecurityType(f) => create_field(f, None::<String>),
            _ => create_field(
                &EditableField {
                    id: None,
                    label: Some("Unknown Field".to_string()),
                    value: EditableFieldString("".to_string()),
                    extensions: None,
                },
                None::<String>,
            ),
        })
        .collect()
}
```

abergs replied:
Done!

abergs committed:
import editablefieldvalue

coroiu reviewed (approved):
Approving (without review) since this is owned by Architecture and Oscar has already reviewed it
